pkg/autoscaler/metrics/stats_scraper_test.go

```diff
@@ -152,47 +181,149 @@ func TestNewServiceScraperWithClientErrorCases(t *testing.T) {
 	}
 }
 
-func TestScrapeReportStatWhenAllCallsSucceed(t *testing.T) {
+func TestPodDirectScrapeSuccess(t *testing.T) {
+	// Easier to reset, than to delete pods.
+	fake.KubeClient = fakek8s.NewSimpleClientset()
+	fake.KubeInformer = kubeinformers.NewSharedInformerFactory(fake.KubeClient, 0)
+
 	client := newTestScrapeClient(testStats, []error{nil})
-	scraper, err := serviceScraperForTest(client)
+	scraper, err := serviceScraperForTest(client, true)
 	if err != nil {
-		t.Fatalf("serviceScraperForTest=%v, want no error", err)
+		t.Fatal("serviceScraperForTest:", err)
 	}
-
-	// Make an Endpoints with 3 pods.
 	fake.Endpoints(3, fake.TestService)
+	makePods(3)
+	now := time.Now()
 
-	// Scrape will set a timestamp bigger than this.
+	got, err := scraper.Scrape(defaultMetric.Spec.StableWindow)
+	if err != nil {
+		t.Fatal("Unexpected error from scraper.Scrape():", err)
+	}
+
+	if got.Time.Before(now) {
+		t.Errorf("stat.Time=%v, want greater than %v", got.Time, now)
+	}
+	if !scraper.podsAddressable {
+		t.Error("PodAddressable switched to false")
+	}
+}
+
+func TestPodDirectScrape1Failure(t *testing.T) {
+	fake.KubeClient = fakek8s.NewSimpleClientset()
+	fake.KubeInformer = kubeinformers.NewSharedInformerFactory(fake.KubeClient, 0)
+
+	// For 5 pods, we need 4 successes.
+	client := newTestScrapeClient(testStats, []error{nil, nil, errors.New("okay"), nil, nil})
+	scraper, err := serviceScraperForTest(client, true)
+	if err != nil {
+		t.Fatal("serviceScraperForTest:", err)
+	}
+	fake.Endpoints(5, fake.TestService)
+	makePods(5)
 	now := time.Now()
+
 	got, err := scraper.Scrape(defaultMetric.Spec.StableWindow)
 	if err != nil {
-		t.Fatalf("Unexpected error from scraper.Scrape(): %v", err)
+		t.Fatal("Unexpected error from scraper.Scrape():", err)
 	}
 
 	if got.Time.Before(now) {
-		t.Errorf("stat.Time=%v, want bigger than %v", got.Time, now)
+		t.Errorf("stat.Time=%v, want greater than %v", got.Time, now)
 	}
-	if got.PodName != scraperPodName {
-		t.Errorf("stat.PodName=%v, want %v", got.PodName, scraperPodName)
+	// Checking one of the metrics is enough here.
+	if got.AverageConcurrentRequests != 20.0 {
+		t.Errorf("stat.AverageConcurrentRequests=%v, want %v",
+			got.AverageConcurrentRequests, 20.0)
 	}
-	// (3.0 + 5.0 + 3.0) / 3.0 * 3 = 11
-	if got.AverageConcurrentRequests != 11.0 {
+
+	if !scraper.podsAddressable {
+		t.Error("PodAddressable switched to false")
+	}
+}
+
+func TestPodDirectScrapeAllFail(t *testing.T) {
+	fake.KubeClient = fakek8s.NewSimpleClientset()
+	fake.KubeInformer = kubeinformers.NewSharedInformerFactory(fake.KubeClient, 0)
+
+	// For 5 pods, we need 4 successes.
```

Comment:
Actually you're creating 4 pods. This comment doesn't match.

Reply:
Removed the comment.